lib/libutils/ext/mempool.c

```diff
@@ -61,26 +62,28 @@ struct mempool {
 	void (*release_mem)(void *ptr, size_t size);
 	struct mutex mu;
 	struct condvar cv;
-	size_t count;
+	struct refcount refc;
 	int owner;
 #endif
 };
 
 static void get_pool(struct mempool *pool __maybe_unused)
 {
 #if defined(__KERNEL__)
-	mutex_lock(&pool->mu);
```

Comment:
Is `mutex_lock()`/`mutex_unlock()` much slower than using `refcount_*` in this case?

Reply:
Yes, I think this is why libmpa was improved quite a bit too. Before I started experimenting with this the MPI performance wasn't that impressive.

Reply:
Hmmm, OK, then perhaps there is something wrong with the mutex fast path? Or perhaps it's the debug code? Anyway, the change is certainly fine by itself.

Reply:
The problem with the fast path is that it does too much, but still it can never become as good as a single `refcount_inc()`.

Reply:
Sure.